annotation/src/main/java/org/cbioportal/genome_nexus/annotation/web/AnnotationController.java

```diff
@@ -155,41 +168,42 @@ public AnnotationController(VariantAnnotationService variantAnnotationService,
 		return variantAnnotations;
 	}
 
-    @ApiOperation(value = "Retrieves VEP annotation for the provided list of variants",
-        nickname = "postVariantAnnotation")
-    @RequestMapping(value = "/hgvs",
-        method = RequestMethod.POST,
+    @ApiOperation(value = "Retrieves VEP annotation for the provided variant",
+        nickname = "fetchVariantAnnotationGET")
+    @RequestMapping(value = "/annotation/{variant:.+}",
+        method = RequestMethod.GET,
         produces = "application/json")
-    public List<VariantAnnotation> postVariantAnnotation(
-        @RequestParam
-        @ApiParam(value="Comma separated list of variants. For example X:g.66937331T>A,17:g.41242962->GA",
-            required = true,
-            allowMultiple = true)
-        List<String> variants,
+    public VariantAnnotation fetchVariantAnnotationGET(
+        @PathVariable
+        @ApiParam(value="Variant. For example 17:g.41242962->GA",
+            required = true)
+            String variant,
         @RequestParam(required = false)
         @ApiParam(value="Isoform override source. For example uniprot",
             required = false)
-        String isoformOverrideSource,
+            String isoformOverrideSource,
         @RequestParam(required = false)
         @ApiParam(value="Comma separated list of fields to include (case-sensitive!). " +
             "For example: hotspots,mutation_assessor", required = false, defaultValue = "hotspots,mutation_assessor")
             List<String> fields)
     {
+        EnrichmentService postEnrichmentService = this.initPostEnrichmentService(isoformOverrideSource, fields);
 
-        return getVariantAnnotation(variants, isoformOverrideSource, fields);
+        VariantAnnotation annotation = getVariantAnnotation(variant);
+
+        if (annotation != null)
+        {
+            postEnrichmentService.enrichAnnotation(annotation);
+        }
+
+		return annotation;
     }
 
-    @ApiOperation(value = "Retrieves hotspot annotation for the provided list of variants",
-        nickname = "getHotspotAnnotation")
-    @ApiResponses(value = {
-        @ApiResponse(code = 200, message = "Success",
-            response = Hotspot.class,
-            responseContainer = "List"),
-        @ApiResponse(code = 400, message = "Bad Request")
-    })
-    @RequestMapping(value = "/cancer_hotspots/{variants:.+}",
-        method = RequestMethod.GET,
-        produces = "application/json")
+//    @ApiOperation(value = "Retrieves hotspot annotation for the provided list of variants",
+//        nickname = "getHotspotAnnotation")
+//    @RequestMapping(value = "/cancer_hotspots/{variants:.+}",
```

Comment:
If there we are dropping the /cancer_hotspots endpoint, shouldn't we move the method "getHotspotAnnotation()" out of the controller and into a service layer? Also, if the function is not mapped to an endpoint with RequestMapping, I wonder what the other java annotations (\@PathVariable \@ApiParam) do? Maybe there is still a default endpoint mapping for this which is not requested?

Reply:
Actually, I am working on a separate internal `CancerHotspotsController` for this endpoint.